src/utils/create-broadcast.js

```diff
@@ -7,29 +7,36 @@
 
 export type Broadcast = {
   publish: (value: mixed) => void,
-  subscribe: (listener: (currentValue: mixed) => void) => () => void
+  subscribe: (listener: (currentValue: mixed) => void) => number,
+  unsubscribe: (number) => void,
 }
 
-const createBroadcast = (initialValue: mixed): Broadcast => {
-  let listeners = []
-  let currentValue = initialValue
+const createBroadcast = (initialState: mixed): Broadcast => {
+  const listeners = {}
+  let id = 0
+  let state = initialState
 
-  return {
-    publish(value: mixed) {
-      currentValue = value
-      listeners.forEach(listener => listener(currentValue))
-    },
-    subscribe(listener) {
-      listeners.push(listener)
+  const publish = (nextState: mixed) => {
+    state = nextState
+    const listenerKeys = Object.keys(listeners)
+    for (let i = 0; i < listenerKeys.length; i += 1) {
```

Comment:
Depends heavily on the browser. For iOS safari, for-in wins by 10%. Probably more without the hasOwnProperty check. 

I'd probably code to make the slowest browsers faster vs targeting desktop ones.